DIPs/DIP1005.md

```diff
@@ -0,0 +1,781 @@
+# Dependency-Carrying Declarations
+
+| Field           | Value                                                      |
+|-----------------|------------------------------------------------------------|
+| DIP:            | 1005                                                       |
+| Author:         | Andrei Alexandrescu (andrei@erdani.com)                    |
+| Implementation: | n/a                                                        |
+| Status:         | Draft                                                      |
+
+## Abstract
+
+A Dependency-Carrying Declaration is a D declaration that does not require any `import` declaration to be present outside of it. Such declarations encapsulate their own dependencies, which makes dependency relationships more fine-grained than traditional module- and package-level dependencies.
+
+Currently D allows definitions to carry their own dependencies by means of the recently-added scoped `import` declarations feature. However, this is not possible with symbols that are present in the symbol declaration itself, for example as function parameter types or template constraints. The limitation reduces the applicability and power of scoped imports. This DIP proposes a language addition called "inline import", which allows any function and aggregate D declarations to be transformed into a Dependency-Carrying Declaration.
```

Comment:
What does `recently-added scoped import declarations feature` refer to exactly? Please provide spec, or changelog, or at least a PR link.